### PR TITLE
Fixed #order for memory relation

### PR DESCRIPTION
**Pull request by nepalez**
Allowed ordering by multiple keys when some values are empty

This is #226 selected to the separate branch

### CODE REVIEW

File: lib/rom/memory/dataset.rb [hunk 1]
```diff
@@ -67,6 +79,14 @@ def delete(tuple)
...
         self
       end
+
+      private
+
+      # Compares two values, that are either comparable, or can be nils
+      def __compare(a, b, nils_first)
+        return a <=> b unless a.nil? ^ b.nil?
+        (nils_first ^ b.nil?) ? -1 : 1
+      end
```
Comment (solnic) [inline lib/rom/memory/dataset.rb]: if we really need a private method please use `__compare__`.

Comment (nepalez) [inline lib/rom/memory/dataset.rb]: Changed

**Comment by solnic** [comment]
This now doesn't merge because of a conflict with CHANGELOG + specs are failing :/

**Comment by nepalez** [comment]
D'oh! Will see it

**Pull request merged**